src/scene/camera.js

```diff
@@ -125,8 +400,8 @@ Object.assign(Camera.prototype, {
         }
 
         if (this._projMatDirty || this._viewMatDirty || this._viewProjMatDirty) {

```

Comment:
could this go into a viewProjMat getter and then shared with the similar block in screenToWorld()?

Reply:
Yes, you're right. The matrix calculation could be refactored too. This PR is already so big though and I'm nervous about making more changes here and would rather do a follow up re-factor. 😄  Let's let the dust settle from this and then take another look.